### DISCUSSION TITLE
Random Hunts/Species/Weapons

### DISCUSSION THREAD

**Discussion by Rancher17** [janstehno/cotw-companion#69] [Ideas]
I'd like to suggest a page with some options for randomized hunts:

1. Random Reserve
2. Random Animal Species (any reserve)
3. Random Reserve and a random animal from that reserve
4. Random Weapon category (Rifle, Handgun, Shotgun, Bow)
5. Random Weapon (specific, including DLC weapons)
6. Random ammo type (this is different than "Random Weapon"; for example, if you rolled ".22LR Truncated Cone", you could use either the Virant .22LR rifle or the Andersson .22LR handgun)
7. Random Animal + random ethical ammo type for that animal
8. a complete Random Hunt (Reserve, Animal, and ethical Weapon).

It would be cool if there were some sort of animation (spinning list like on slot-machines for example), but this is not critical.

Another feature could be some random hunt restraints such as (for example):

- Dogs not allowed
- Nighttime only
- Callers not allowed
- Nightvision not allowed
- Only tritium and iron sights allowed
- Treestands not allowed
- Tripods not allowed
- ATVs not allowed
- Fast-travel not allowed
- Must hunt at least 80m from a water source
- Only base-game equipment allowed (no DLC equipment)
- etc.
- 
I am sure there are many others.

**Comment by janstehno** [comment]
I like it! Already suggested loadout generator is planned for a long time now. This is even better and will be implemented for sure.

**Comment by janstehno** [comment]
So I have experimented with this feature a bit. Here is how it could look like:

### Every option is allowed to be generated randomly
<img src="https://github.com/janstehno/cotw-companion/assets/72458808/3d94fe0f-6d6a-4de6-9b33-ff4e119583f9" width=250>

### Options for random reserve, maximum distance, usage of an atv, and fast travel option are disabled
<img src="https://github.com/janstehno/cotw-companion/assets/72458808/d13b7faf-15b4-479c-916a-ecca70507e86" width=250>

The feature will not have any animations as I have no idea how I would implement them. Also the feature could be less performant because of the animations.

The app will remember enabled and disabled options. However it won't remember any generated hunting pass.

Let me know what do you think about it. I tried to keep it light and easy to understand.

**Reply by Rancher17** [reply]
I think it looks great! One way to shorten the options is to reword them slightly:

Hunting dog allowed
Callers/Scents allowed
Scopes allowed (_or_ Iron-sights only)
ATV allowed

_etc_.

The green/red icon makes it clear what is intended (Yes/No).

I would suggest adding "Night-hunt only (20.00-5.00)"

**Reply by janstehno** [reply]
Added "Day time hunting (5-20)" instead of night one. Makes more sense in terms of "allowing something". Also adjusted the other options, they were too complicated indeed. Here is a 20s showcase of how it could look like.

### Video showcase
https://github.com/janstehno/cotw-companion/assets/72458808/d5f44424-ef52-476b-aed8-0c5e89dca617

**Comment by janstehno** [comment]
Added in the version 1.10.3
